src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,60 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int DEFAULT_RADIUS = 10;
+    private static final int MIN_SIZE = 10;
+    private static final int MAX_SIZE = 20;
+    private static final int DIFF = MAX_SIZE - MIN_SIZE;
+    private static final String DEFAULT_COLOR = "white";
+    private Random random = new Random();
+    private ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        switch (random.nextInt(5)) {
+            case 0:
+                return getRandomCircle();
+            case 1:
+                return getRandomIsoscelesTrapezoid();
+            case 2:
+                return getRandomRectangle();
+            case 3:
+                return getRandomRightTriangle();
+            case 4:
+            default:
+                return getRandomSquare();
+        }
+    }
+
+    public Figure getDefaultFigure() {
+        return new Circle(DEFAULT_COLOR, DEFAULT_RADIUS);
+    }
+
+    private Figure getRandomCircle() {
+        return new Circle(colorSupplier.getRandomColor(), random.nextInt(DIFF) + MIN_SIZE);
```

Comment:
Why make it so complicated and add 10 to a random number every time? Just pass the maximum desired number to nextInt at once. And delete DIFF & MIN_SIZE

Reply:
I just wanted to make sure that trapezoid area would never be NaN. I'll fix it.